uwp/uwp_main.cpp

```diff
@@ -638,6 +681,57 @@ void App::OnPackageInstalling(PackageCatalog^ sender, PackageInstallingEventArgs
 	}
 }
 
+void App::ParseProtocolArgs(Windows::ApplicationModel::Activation::IActivatedEventArgs^ args, int *argc, std::vector<char*> *argv, std::vector<std::string> *argvTmp)
+{
+	argvTmp->clear();
+	argv->clear();
+
+	if (args->Kind == ActivationKind::Protocol)
+	{
+		ProtocolActivatedEventArgs^ protocolArgs = dynamic_cast<Windows::ApplicationModel::Activation::ProtocolActivatedEventArgs^>(args);
+		Windows::Foundation::WwwFormUrlDecoder^ query = protocolArgs->Uri->QueryParsed;
+
+		//if RetroArch UWP app is started using protocol with argument "launchOnExit", this gives an option to launch another app on RA exit,
+		//making it easy to integrate RA with other UWP frontends
+		try
+		{
+			m_launchOnExit = query->GetFirstValueByName("launchOnExit");
+		}
+		catch (Platform::InvalidArgumentException^ e)
+		{
+			//nothing to do if named parameter doesn't exist
+		}
+
+		try
```

Comment:
is there no way to parse this without using try catch?

Reply:
Yes, we can definitely remove try catch and just go through each argument.

Reply:
Removed try catch blocks, not using GetFirstValueByName method.